### PR TITLE
Replaced LRUMap with a Guava Cache

### PR DESCRIPTION
**Pull request by lynyus**
This is intended as a fix for #2546.

Note that the Guava Caches would allow for automated reloading of the entries, but this is not undertaken in this implementation. Instead everything should work as before with the LRUMap.

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [x] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

**Comment by stefan-kolb** [comment]
Thanks ❤️ 
Can you please also change the external-libraries.txt and remove commons from there?

**Comment by stefan-kolb** [comment]
Also small issue:
```
checkstyleMain[ant:checkstyle] [ERROR] /home/travis/build/JabRef/jabref/src/main/java/net/sf/jabref/logic/pdf/FileAnnotationCache.java:6: Wrong order for 'net.sf.jabref.model.database.BibDatabaseContext' import. [ImportOrder]
[ant:checkstyle] [ERROR] /home/travis/build/JabRef/jabref/src/main/java/net/sf/jabref/logic/pdf/FileAnnotationCache.java:10: Wrong order for 'java.util.HashMap' import. [ImportOrder]
```

File: src/main/java/net/sf/jabref/logic/pdf/FileAnnotationCache.java [hunk 1]
```diff
@@ -1,43 +1,58 @@
...
+    private final static int CACHE_SIZE = 10;
+    //the inner list holds the annotations per file, the outer collection maps this to a BibEntry.
+    LoadingCache<BibEntry, Map<String, List<FileAnnotation>>> annotationCache;
     BibDatabaseContext bibDatabaseContext;
 
     public FileAnnotationCache(BibDatabaseContext bibDatabaseContext) {
         this.bibDatabaseContext = bibDatabaseContext;
-        annotationCache  = new LRUMap<>(CACHE_SIZE);
+        annotationCache = CacheBuilder.newBuilder().maximumSize(CACHE_SIZE).build(new CacheLoader<BibEntry, Map<String, List<FileAnnotation>>>() {
+
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/pdf/FileAnnotationCache.java]: You may remove this line

**Comment by stefan-kolb** [comment]
Checkstyle help: https://github.com/JabRef/jabref/tree/master/config

**Comment by lynyus** [comment]
thanks for the feedback, there is quite a potential to improve stuff with these caches. Maybe I'll have a look at the big picture there later.

**Pull request merged**